src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java

```diff
@@ -151,6 +149,10 @@ else if (structureBlock instanceof BlockStairs && structureBlockState.equals(wor
             {
                 return true;
             }
+            else if((structureBlock == Blocks.DIRT || structureBlock == Blocks.GRASS) && (worldBlock == Blocks.DIRT || worldBlock == Blocks.GRASS))
+            {
```

Comment:
Honestly, if there is a way to better check that I'm all in.
I just didn't find any better way.